dbms/src/TestUtils/ExecutorTestUtils.cpp

```diff
@@ -0,0 +1,130 @@
+    dag_context_ptr->log = Logger::get("executorTest");
+}
+
+void ExecutorTest::SetUpTestCase()
+{
+    try
+    {
+        DB::registerFunctions();
+        DB::registerAggregateFunctions();
+    }
+    catch (DB::Exception &)
+    {
+        // Maybe another test has already registered, ignore exception here.
+    }
+}
+
+void ExecutorTest::initializeClientInfo()
+{
+    context.context.setCurrentQueryId("test");
+    ClientInfo & client_info = context.context.getClientInfo();
+    client_info.query_kind = ClientInfo::QueryKind::INITIAL_QUERY;
+    client_info.interface = ClientInfo::Interface::GRPC;
+}
+
+void ExecutorTest::executeInterpreter(const String & expected_string, const std::shared_ptr<tipb::DAGRequest> & request, size_t concurrency)
+{
+    DAGContext dag_context(*request, "interpreter_test", concurrency);
+    context.context.setDAGContext(&dag_context);
+    // Currently, don't care about regions information in interpreter tests.
+    DAGQuerySource dag(context.context);
+    auto res = executeQuery(dag, context.context, false, QueryProcessingStage::Complete);
+    FmtBuffer fb;
+    res.in->dumpTree(fb);
+    ASSERT_EQ(Poco::trim(expected_string), Poco::trim(fb.toString()));
+}
+
+static Block mergeBlocks(Blocks blocks)
+{
+    if (blocks.empty())
+        return {};
+
+    Block sample_block = blocks.back();
+    std::vector<MutableColumnPtr> actual_cols;
+    for (const auto & column : sample_block.getColumnsWithTypeAndName())
+    {
+        actual_cols.push_back(column.type->createColumn());
+    }
+    for (const auto & block : blocks)
+    {
+        for (size_t i = 0; i < block.columns(); ++i)
+        {
+            for (size_t j = 0; j < block.rows(); ++j)
+            {
+                actual_cols[i]->insert((*(block.getColumnsWithTypeAndName())[i].column)[j]);
+            }
+        }
+    }
+
+    ColumnsWithTypeAndName actual_columns;
+    for (size_t i = 0; i < actual_cols.size(); ++i)
+        actual_columns.push_back({std::move(actual_cols[i]), sample_block.getColumnsWithTypeAndName()[i].type, sample_block.getColumnsWithTypeAndName()[i].name, sample_block.getColumnsWithTypeAndName()[i].column_id});
+    return Block(actual_columns);
+}
+
+void readBlock(BlockInputStreamPtr stream, const ColumnsWithTypeAndName & expect_columns)
+{
+    Blocks actual_blocks;
+    Block except_block(expect_columns);
+    stream->readPrefix();
+    while (auto block = stream->read())
+    {
+        actual_blocks.push_back(block);
+    }
+    stream->readSuffix();
+    Block actual_block = mergeBlocks(actual_blocks);
+    ASSERT_BLOCK_EQ(except_block, actual_block);
+}
+
+void ExecutorTest::executeStreams(const std::shared_ptr<tipb::DAGRequest> & request, std::unordered_map<String, ColumnsWithTypeAndName> & source_columns_map, const ColumnsWithTypeAndName & expect_columns, size_t concurrency)
+{
+    DAGContext dag_context(*request, "interpreter_test", concurrency);
```

Comment:
You rename InterpreterTestUtils.cpp to ExecutorTestUtil.cpp , If here should be changed accordingly.